File: solutions/cpp/pangram/1/pangram.cpp
#include "pangram.h"

#include <bitset>
#include <cctype>

namespace pangram {

bool is_pangram(const std::string& sentence) {
    std::bitset<26> alphabet{};

    for (unsigned char c : sentence) {
        if (std::isalpha(c)) {
            alphabet.set(std::tolower(c) - 'a');
        }
    }

    return alphabet.all();
}

} // namespace pangram

File: solutions/csharp/pangram/1/Pangram.cs
using System;
using System.Linq;
using System.Text.RegularExpressions;

public static class Pangram
{
    public static bool IsPangram(string input)
    {
        int CharacterIndex(char character) => character - 'a';

        string valid_chars = "abcdefghijklmnopqrstuvwxyz";
        bool[] hits = new bool[valid_chars.Length];

        string stripped_input = Regex.Replace(input.ToLower(), "[^a-z]", "");
        foreach (char c in stripped_input)
        {
            hits[CharacterIndex(c)] = true;
        }

        return hits.Aggregate(true, (a, b) => a && b);
    }
}
